test/tools/unified-spec-runner/match.ts

```diff
@@ -139,78 +139,96 @@ export function resultCheck(
   expected: Document | number | string | boolean,
   entities: EntitiesMap,
   path: string[] = [],
-  depth = 0
+  isRootDocument = true
 ): void {
+  function checkNestedDocuments(key: string, value: any) {
+    if (key === 'sort') {
+      // TODO: This is a workaround that works because all sorts in the specs
+      // are objects with one key; ideally we'd want to adjust the spec definitions
+      // to indicate whether order matters for any given key and set general
+      // expectations accordingly (see NODE-3235)
+      expect(Object.keys(value)).to.have.lengthOf(1);
+      expect(actual[key]).to.be.instanceOf(Map);
+      expect(actual[key].size).to.equal(1);
+      const expectedSortKey = Object.keys(value)[0];
+      expect(actual[key]).to.have.all.keys(expectedSortKey);
+      const objFromActual = { [expectedSortKey]: actual[key].get(expectedSortKey) };
+      resultCheck(objFromActual, value, entities, path, false);
+    } else {
+      resultCheck(actual[key], value, entities, path, false);
+    }
+  }
+
   if (typeof expected === 'object' && expected) {
     // Expected is an object
     // either its a special operator or just an object to check equality against
 
     if (isSpecialOperator(expected)) {
       // Special operation check is a base condition
       // specialCheck may recurse depending upon the check ($$unsetOrMatches)
-      specialCheck(actual, expected, entities, path, depth);
+      specialCheck(actual, expected, entities, path, false);
```

Comment:
should this be `isRootDocument` instead of `false`? There's a bunch of failing tests, and this is the logic that has changed since the last commit